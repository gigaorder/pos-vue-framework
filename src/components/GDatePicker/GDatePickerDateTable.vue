<script>
  import { createRange } from '../../utils/helpers';
  import { pad, createNativeLocaleFormatter, monthChange } from './util'
  import { computed, createElement as h } from '@vue/composition-api';
  import GDatePickerTable from './mixins/date-picker-table'

  export default {
    name: 'GDatePickerDateTable',
    props: {
      // GDatePickerTable
      allowedDates: null,
      current: String,
      disabled: Boolean,
      format: null,
      events: {
        type: [Array, Function, Object],
        default: () => null,
      },
      eventColor: {
        type: [Array, Function, Object, String],
        default: () => 'warning',
      },
      min: String,
      max: String,
      readonly: Boolean,
      scrollable: Boolean,
      tableDate: {
        type: String,
        required: true,
      },
      value: [String, Array],

      // Colorable
      color: [String],

      // GDatePickerDateTable
      firstDayOfWeek: {
        type: [String, Number],
        default: 0,
      },
      showWeek: Boolean,
      weekdayFormat: Function
    },
    setup(props, context) {
      const {
        displayedMonth,
        displayedYear,
        genButton,
        genTable
      } = GDatePickerTable(props, context)

      // computed
      const formatter = computed(() => {
        return props.format || createNativeLocaleFormatter(undefined/*TODO: this.currentLocale*/, { day: 'numeric', timeZone: 'UTC' }, { start: 8, length: 2 })
      })

      const weekdayFormatter = computed(() => {
        return props.weekdayFormat || createNativeLocaleFormatter(undefined/*TODO: this.currentLocale*/, { weekday: 'narrow', timeZone: 'UTC' })
      })

      const weekDays = computed(() => {
        const first = parseInt(props.firstDayOfWeek, 10)

        return weekdayFormatter.value
            ? createRange(7).map(i => weekdayFormatter.value(`2017-01-${first + i + 15}`)) // 2017-01-15 is Sunday
            : createRange(7).map(i => ['S', 'M', 'T', 'W', 'T', 'F', 'S'][(i + first) % 7])
      })

      // methods
      function calculateTableDate(delta/*: number*/) {
        return monthChange(props.tableDate, Math.sign(delta || 1))
      }

      function genTHead() {
        const days = weekDays.value.map(day => h('th', day))
        props.showWeek && days.unshift(h('th'))
        return h('thead', genTR(days))
      }

      // Returns number of the days from the firstDayOfWeek to the first day of the current month
      function weekDaysBeforeFirstDayOfTheMonth() {
        const firstDayOfTheMonth = new Date(`${displayedYear.value}-${pad(displayedMonth.value + 1)}-01T00:00:00+00:00`)
        const weekDay = firstDayOfTheMonth.getUTCDay()
        return (weekDay - parseInt(props.firstDayOfWeek) + 7) % 7
      }

      function getWeekNumber() {
        let dayOfYear = [0, 31, 59, 90, 120, 151, 181, 212, 243, 273, 304, 334][displayedMonth.value]
        if (displayedMonth.value > 1 &&
            (((displayedYear.value % 4 === 0) && (displayedYear.value % 100 !== 0)) || (displayedYear.value % 400 === 0))
        ) {
          dayOfYear++
        }
        const offset = (
            displayedYear.value +
            ((displayedYear.value - 1) >> 2) -
            Math.floor((displayedYear.value - 1) / 100) +
            Math.floor((displayedYear.value - 1) / 400) -
            Number(props.firstDayOfWeek)
        ) % 7 // https://en.wikipedia.org/wiki/Zeller%27s_congruence
        return Math.floor((dayOfYear + offset) / 7) + 1
      }

      function genWeekNumber(weekNumber/*: number*/) {
        return h('td', [
          h('small', {
            staticClass: 'g-date-picker-table--date__week',
          }, String(weekNumber).padStart(2, '0')),
        ])
      }

      function genTBody() {
        const children = []
        const daysInMonth = new Date(displayedYear.value, displayedMonth.value + 1, 0).getDate()
        let rows = []
        let day = weekDaysBeforeFirstDayOfTheMonth()
        let weekNumber = getWeekNumber()

        props.showWeek && rows.push(genWeekNumber(weekNumber++))

        while (day--) rows.push(h('td'))
        for (day = 1; day <= daysInMonth; day++) {
          const date = `${displayedYear.value}-${pad(displayedMonth.value + 1)}-${pad(day)}`

          rows.push(h('td', [
            genButton(date, true, 'date', formatter.value),
          ]))

          if (rows.length % (props.showWeek ? 8 : 7) === 0) {
            children.push(genTR(rows))
            rows = []
            day < daysInMonth && props.showWeek && rows.push(genWeekNumber(weekNumber++))
          }
        }

        if (rows.length) {
          children.push(genTR(rows))
        }

        return h('tbody', children)
      }

      function genTR(children) {
        return [h('tr', children)]
      }

      return () => {
        return genTable('g-date-picker-table g-date-picker-table--date', [
          genTHead(),
          genTBody(),
        ], calculateTableDate)
      }
    }
  }
</script>
<style scoped>
</style>
