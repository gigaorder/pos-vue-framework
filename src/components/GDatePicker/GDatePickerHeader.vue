<template>
  <div :class="datePickerHeaderClasses">
    <div :class="datePickerHeaderValueClasses">
      <transition :name="headerTransitionName">
        <div :key="value" :class="headerData.class" :style="headerData.style">
          <button type="button" v-on:click="onHeaderClicked">
            <template v-if="defaultSlotIsAvailable">
              <slot></slot>
            </template>
            <template v-else>
              {{ headerFormatter(value) }}
            </template>
          </button>
        </div>
      </transition>
    </div>
    <button :disabled="prevBtnDisabled" class="g-date-picker-header__prev-button" v-on:click="prevClick">
<!--      <i class="material-icons">mdi-chevron-left</i>-->
    </button>
    <button :disabled="nextBtnDisabled" class="g-date-picker-header__next-button" v-on:click="nextClick">
<!--      <i class="material-icons">mdi-chevron-right</i>-->
    </button>
  </div>
</template>

<script>
  import { computed, reactive, watch } from '@vue/composition-api'
  import { monthChange, createNativeLocaleFormatter, TRANSITION_NAMES } from './utils'
  import { setTextColor } from '../../mixins/colorable'
  import GButton from '../GButton/GButton'

  // clss
  const DATE_PICKER_HEADER_CLS = {
    HEADER: 'g-date-picker-header',
    HEADER_DISABLED: 'g-date-picker-header--disabled',
    HEADER_VALUE: 'g-date-picker-header__value',
    HEADER_VALUE_DISABLED: 'g-date-picker-header__value--disabled'
  }

  const NAV = {
    PREV: -1,
    NEXT: 1
  }

  export const EVENT_NAMES = {
    TOGGLE: 'toggle',
    INPUT: 'input'
  }

  export default {
    name: 'GDatePickerHeader',
    components: { GButton },
    props: {
      disabled: Boolean,
      readonly: Boolean,
      min: String,
      max: String,
      value: {
        type: [Number, String],
        required: true,
      },
      format: Function,
    },
    setup(props, context) {
      // header transition
      const state = reactive({ isReversing: false })
      const headerTransitionName = computed(() => (state.isReversing) ? TRANSITION_NAMES.REVERSE_TAB : TRANSITION_NAMES.TAB)
      watch(() => props.value, (newVal, oldVal) => state.isReversing = newVal < oldVal, { lazy: true, flush: 'pre' })
      // header value
      const headerFormatter = computed(() => {
        if (props.format) {
          return props.format
        } else if (String(props.value).split('-')[1]) {
          return createNativeLocaleFormatter(undefined, { month: 'long', year: 'numeric', timeZone: 'UTC' }, { length: 7 })
        } else {
          return createNativeLocaleFormatter(undefined, { year: 'numeric', timeZone: 'UTC' }, { length: 4 })
        }
      })
      const datePickerHeaderClasses = computed(() => {
        return {
          [DATE_PICKER_HEADER_CLS.HEADER]: true,
          [DATE_PICKER_HEADER_CLS.HEADER_DISABLED]: props.disabled
        }
      })
      const datePickerHeaderValueClasses = computed(() => {
        return {
          [DATE_PICKER_HEADER_CLS.HEADER_VALUE]: true,
          [DATE_PICKER_HEADER_CLS.HEADER_VALUE_DISABLED]: props.disabled
        }
      })
      const headerData = computed(() => setTextColor(props.color, {}))
      const defaultSlotIsAvailable = context.slots.default !== undefined
      // header event handler
      const onHeaderClicked = () => context.emit(EVENT_NAMES.TOGGLE)


      // Prev button
      const prevBtnDisabled = computed(() => props.disabled || (props.min && calculateChange(NAV.PREV) < props.min))
      const prevClick = () => context.emit(EVENT_NAMES.INPUT, calculateChange(NAV.PREV))

      // Next button
      const nextBtnDisabled = computed(() => props.disabled || (props.max && calculateChange(NAV.NEXT) < props.max))
      const nextClick = () => context.emit(EVENT_NAMES.INPUT, calculateChange(NAV.NEXT))

      //
      function calculateChange(sign/*: number*/) {
        const [year, month] = String(props.value).split('-').map(Number)
        if (month == null) {
          return `${year + sign}`
        } else {
          return monthChange(props.value, sign)
        }
      }


      return {
        datePickerHeaderClasses,
        datePickerHeaderValueClasses,
        headerData,
        headerTransitionName,
        defaultSlotIsAvailable,
        onHeaderClicked,
        headerFormatter,
        //
        prevBtnDisabled,
        prevClick,
        //
        nextBtnDisabled,
        nextClick,

      }
    }
  }
</script>
<style scoped>
</style>
