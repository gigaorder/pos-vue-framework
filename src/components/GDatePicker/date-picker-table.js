import { computed } from '@vue/composition-api';

export default (props, context) => {
  const displayedMonth = computed(() => {
    return Number(props.tableDate.split('-')[1]) - 1
  })
  const displayedYear = computed(() => {
    return Number(props.tableDate.split('-')[0])
  })

  function genButtonClasses(isAllowed, isFloating, isSelected, isCurrent) {
    return {
      'g-size--default': !isFloating,
      'g-btn--active': isSelected,
      'g-btn--flat': !isAllowed || props.disabled,
      'g-btn--text': isSelected === isCurrent,
      'g-btn--rounded': isFloating,
      'g-btn--disabled': !isAllowed || props.disabled,
      'g-btn--outlined': isCurrent && !isSelected,
    }
  }

  return {
    genButtonClasses,
    displayedMonth,
    displayedYear,
  }
}
