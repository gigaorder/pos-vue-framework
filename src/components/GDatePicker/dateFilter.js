/**
 * Check if provided value is in range
 * @param value
 * @param min
 * @param max
 * @param isValueInRangeCustomFn
 * @returns {boolean|*}
 */
export const isValueInRange = (value, min, max) => {
  return (!min || value >= min) && (!max || value <= max)
}

export function isDateAllowed(value, filterFn) {
  if (typeof filterFn !== 'function')
    return true

  return filterFn(value)
}

export function dateFilter(props) {
  return function isDateAvailable(value) {
    return isDateAllowed(value, props.allowedDates) && isValueInRange(value, props.min, props.max)
  }
}
