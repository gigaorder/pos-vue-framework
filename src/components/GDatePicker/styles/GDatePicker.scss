@mixin transition-default() {
  &-enter-active,
  &-leave-active {
    transition: $primary-transition;
  }

  &-move {
    transition: transform .6s;
  }
}

.picker-transition,
.picker-reverse-transition {
  &-enter-active,
  &-leave-active {
    transition: .3s map-get($transition, 'linear-out-slow-in');
  }

  &-enter,
  &-leave-to {
    opacity: 0;
  }

  &-leave,
  &-leave-active,
  &-leave-to {
    position: absolute !important;
  }
}

.picker-transition {
  &-enter {
    transform: translate(0, 100%);
  }

  &-leave-to {
    transform: translate(0, -100%);
  }
}

.picker-reverse-transition {
  &-enter {
    transform: translate(0, -100%);
  }

  &-leave-to {
    transform: translate(0, 100%);
  }
}

.picker-title-transition {
  &-enter-to,
  &-leave {
    transform: translate(0, 0);
  }

  &-enter {
    transform: translate(-100%, 0);
  }

  &-leave-to {
    opacity: 0;
    transform: translate(100%, 0);
  }

  &-leave,
  &-leave-to,
  &-leave-active {
    position: absolute !important;
  }
}

.tab-transition {
  &-enter {
    transform: translate(100%, 0);
  }

  &-leave, &-leave-active {
    position: absolute;
    top: 0;
  }

  &-leave-to {
    position: absolute;
    transform: translate(-100%, 0);
  }
}

.tab-reverse-transition {
  &-enter {
    transform: translate(-100%, 0);
  }

  &-leave, &-leave-to {
    top: 0;
    position: absolute;
    transform: translate(100%, 0);
  }
}

.fade-transition {
  @include transition-default();

  &-enter, &-leave-to {
    opacity: 0 !important;
  }
}
