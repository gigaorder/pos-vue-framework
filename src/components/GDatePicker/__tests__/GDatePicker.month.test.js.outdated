import { createLocalVue, mount } from '@vue/test-utils'
import plugin from '@vue/composition-api'
import GDatePicker from '../GDatePicker';

describe('GDatePicker.month.js', () => {
  let mountFunction

  const localVue = createLocalVue()
  localVue.use(plugin)

  beforeEach(() => {
    mountFunction = (options) => {
      return mount(GDatePicker, {
        localVue,
        ...options
      })
    }
  })

  it('should emit input event on month click', async () => {
    const cb = jest.fn()
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
      },
    })

    wrapper.vm.$on('input', cb)

    const target = wrapper.findAll('.g-date-picker-table--month button').at(0)
    target.trigger('click')
    expect(cb).toHaveBeenCalledWith('2013-01')
  })

  it('should be scrollable', (done) => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        scrollable: true,
      },
    })

    wrapper.findAll('.g-date-picker-table--month').at(0).trigger('wheel')
    wrapper.vm.$nextTick(() => {
      expect(wrapper.vm.state.tableDate).toBe('2014')
      done()
    })
  })

  it('should match snapshot with pick-month prop', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05-07',
        type: 'month',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with allowed dates as array', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
        allowedDates: value => ['2013-01', '2013-03', '2013-05', '2013-07'].includes(value),
      },
    })

    expect(wrapper.findAll('.g-date-picker-table--month tbody').at(0).html()).toMatchSnapshot()
  })

  it('should match snapshot with month formatting functions', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11-01',
        type: 'month',
        monthFormat: date => `(${date.split('-')[1]})`,
      },
    })

    const target = wrapper.findAll('.g-date-picker-table--month tbody').at(0)
    expect(target.html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker & header', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'primary',
        headerColor: 'orange darken-1',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match snapshot with colored picker', () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11-01',
        color: 'orange darken-1',
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should match change month when clicked on header arrow buttons', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11',
        type: 'month',
      },
    })

    const leftButton = wrapper.findAll('.g-date-picker-header__prev-button').at(0)
    const rightButton = wrapper.findAll('.g-date-picker-header__next-button').at(0)

    leftButton.trigger('click')
    expect(wrapper.vm.state.tableDate).toBe('2004')

    rightButton.trigger('click')
    expect(wrapper.vm.state.tableDate).toBe('2005')
  })

  it('should match change active picker when clicked on month button', () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2005-11-01',
        type: 'month',
      },
    })

    const button = wrapper.findAll('.g-date-picker-header__value button').at(0)

    button.trigger('click')
    expect(wrapper.vm.state.activePicker).toBe('year')
  })

  it('should select year', async () => {
    const wrapper = mountFunction({
      propsData: {
        type: 'month',
        value: '2005-11',
      },
    })

    wrapper.vm.state.activePicker = 'year'
    wrapper.findAll('.g-date-picker-years li.active + li').at(0).trigger('click')
    expect(wrapper.vm.state.activePicker).toBe('month')
    expect(wrapper.vm.state.tableDate).toBe('2004')
  })

  it('should set the table date when value has changed', () => {
    const wrapper = mountFunction({
      propsData: {
        value: null,
        type: 'month',
      },
    })

    wrapper.setProps({ value: '2005-11' })
    expect(wrapper.vm.state.tableDate).toBe('2005')
  })

  it('should display translated title', async () => {
    const wrapper = mountFunction({
      propsData: {
        multiple: true,
        type: 'month',
        value: ['2013-05'],
      },
    })

    expect(wrapper.find('.g-date-picker-title__date').text()).toBe('1 selected')

    wrapper.setProps({ value: [] })
    expect(wrapper.find('.g-date-picker-title__date').text()).toBe('0 selected')

    wrapper.setProps({
      value: ['2013-05', '2013-06', '2013-07'],
    })
    expect(wrapper.find('.g-date-picker-title__date').text()).toBe('3 selected')
  })

  it('should emit click event', async () => {
    const wrapper = mountFunction({
      propsData: {
        value: '2013-05',
        type: 'month',
      },
    })

    const click = jest.fn()
    wrapper.vm.$on(`click:month`, click)
    wrapper.findAll('.g-date-picker-table--month tbody tr+tr td:first-child button').at(0).trigger('click')
    expect(click).toHaveBeenCalledWith('2013-04')
  })

})
