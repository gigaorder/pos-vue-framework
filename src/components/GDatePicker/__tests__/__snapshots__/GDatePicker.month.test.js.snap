// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GDatePicker.month.js should match snapshot with allowed dates as array 1`] = `
<tbody>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          Jan
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Feb
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          Mar
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Apr
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn g-btn--active"
              style="background-color: rgb(77, 0, 234); border-color: rgb(77, 0, 234);"
      >
        <div class="g-btn__content">
          May
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Jun
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          Jul
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Aug
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Sep
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Oct
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Nov
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              disabled="disabled"
              class="g-btn g-btn--disabled"
      >
        <div class="g-btn__content">
          Dec
        </div>
      </button>
    </td>
  </tr>
</tbody>
`;

exports[`GDatePicker.month.js should match snapshot with colored picker & header 1`] = `
<div class="g-picker">
  <div class="g-picker__title orange darken-1">
    <div class="g-date-picker-title">
      <div class="g-picker__title__btn g-picker__title__btn--active g-date-picker-title__year">
        2005
      </div>
      <div class="g-picker__title__btn g-date-picker-title__date">
        <div name="picker-transition">
          November
        </div>
      </div>
    </div>
  </div>
  <div class="g-picker__body"
       style="width: 290px;"
  >
    <div>
      <div class="g-date-picker-header">
        <div class="g-date-picker-header__value">
          <div name="tab-transition">
            <button type="button">
              2005
            </button>
          </div>
        </div>
        <button class="g-date-picker-header__prev-button">
        </button>
        <button class="g-date-picker-header__next-button">
        </button>
      </div>
      <div class="g-date-picker-table g-date-picker-table--month">
        <table name="tab-transition">
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div class="g-picker__actions">
  </div>
</div>
`;

exports[`GDatePicker.month.js should match snapshot with colored picker 1`] = `
<div class="g-picker">
  <div class="g-picker__title orange darken-1">
    <div class="g-date-picker-title">
      <div class="g-picker__title__btn g-picker__title__btn--active g-date-picker-title__year">
        2005
      </div>
      <div class="g-picker__title__btn g-date-picker-title__date">
        <div name="picker-transition">
          November
        </div>
      </div>
    </div>
  </div>
  <div class="g-picker__body"
       style="width: 290px;"
  >
    <div>
      <div class="g-date-picker-header">
        <div class="g-date-picker-header__value">
          <div name="tab-transition">
            <button type="button">
              2005
            </button>
          </div>
        </div>
        <button class="g-date-picker-header__prev-button">
        </button>
        <button class="g-date-picker-header__next-button">
        </button>
      </div>
      <div class="g-date-picker-table g-date-picker-table--month">
        <table name="tab-transition">
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div class="g-picker__actions">
  </div>
</div>
`;

exports[`GDatePicker.month.js should match snapshot with month formatting functions 1`] = `
<tbody>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (01)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (02)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (03)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (04)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (05)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (06)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (07)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (08)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (09)
        </div>
      </button>
    </td>
  </tr>
  <tr>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (10)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (11)
        </div>
      </button>
    </td>
    <td>
      <button type="button"
              class="g-btn"
      >
        <div class="g-btn__content">
          (12)
        </div>
      </button>
    </td>
  </tr>
</tbody>
`;

exports[`GDatePicker.month.js should match snapshot with pick-month prop 1`] = `
<div class="g-picker">
  <div class="g-picker__title"
       style="background-color: rgb(77, 0, 234); border-color: rgb(77, 0, 234);"
  >
    <div class="g-date-picker-title">
      <div class="g-picker__title__btn g-picker__title__btn--active g-date-picker-title__year">
        2013
      </div>
      <div class="g-picker__title__btn g-date-picker-title__date">
        <div name="picker-transition">
          May
        </div>
      </div>
    </div>
  </div>
  <div class="g-picker__body"
       style="width: 290px;"
  >
    <div>
      <div class="g-date-picker-header">
        <div class="g-date-picker-header__value">
          <div name="tab-transition">
            <button type="button">
              2013
            </button>
          </div>
        </div>
        <button class="g-date-picker-header__prev-button">
        </button>
        <button class="g-date-picker-header__next-button">
        </button>
      </div>
      <div class="g-date-picker-table g-date-picker-table--month">
        <table name="tab-transition">
          <tbody>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jan
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Feb
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Mar
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Apr
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    May
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jun
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Jul
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Aug
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Sep
                  </div>
                </button>
              </td>
            </tr>
            <tr>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Oct
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Nov
                  </div>
                </button>
              </td>
              <td>
                <button type="button"
                        class="g-btn"
                >
                  <div class="g-btn__content">
                    Dec
                  </div>
                </button>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
  <div class="g-picker__actions">
  </div>
</div>
`;
